### PR TITLE
chore: update keyfunc version

### PR DESCRIPTION
**Pull request by MicahParks**
The purpose of this PR is to update the `github.com/MicahParks/keyfunc` package. The primary reason I'm making this PR is to introduce the changes from these releases:
* [Support non-RFC compliant base64url padding](https://github.com/MicahParks/keyfunc/releases/tag/v1.1.0) (Compatibility update.)
* [Check HTTP status code](https://github.com/MicahParks/keyfunc/releases/tag/v1.4.0) (Confirms the HTTP status code is 200. Bug fix.)
* [JSON Web Keys restricted by "use" parameter](https://github.com/MicahParks/keyfunc/releases/tag/v1.5.1) (Better RFC compliance. Could be viewed as a bug fix.)

If there is any interest in the other features added since `v1.0.0`, please see the release page:
https://github.com/MicahParks/keyfunc/releases

For a quick summary of the new `keyfunc.Options`, see the documentation for its fields:
https://pkg.go.dev/github.com/MicahParks/keyfunc#Options

#### Checklist

- [x] run `make test`
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

The Go tests, `go test ./...` pass. I don't have `pnpm` installed, so I have not run those tests.

I'm happy to fill out this checklist tomorrow EDT time, if requested. I know I'll likely need to make a properly formatted commit message, if that can't be done via a merge commit.

Fixes ENG-807

### CODE REVIEW

**Comment by jensneuse** [comment]
Hi @MicahParks thanks for this PR. Can you give us some hints in what state this PR is?
As of now, you've only updated the go mod file. I'm not sure if this was intended.

**Comment by MicahParks** [comment]
> Hi @MicahParks thanks for this PR. Can you give us some hints in what state this PR is?
> As of now, you've only updated the go mod file. I'm not sure if this was intended.

The PR contains all the changes I would like to see in its current state. I simply want to update `keyfunc` and draw attention to the defaults that have changed since the version in this project's `go.mod`.

However, I see there are some requirements around PRs for this repository and I want to respect those requirements, if necessary for this change. I haven't had time to include what's generally required for PRs, so I'll need to update this PR to meet those requirements when I get the time.

Alternatively, if you'd like the PR merged now, feel free to do so, making edits how you see fit. It's a busy week for me 🙂

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/wundergraph/wundergraph
 go 1.18
 
 require (
-	github.com/MicahParks/keyfunc v1.5.1
+	github.com/MicahParks/keyfunc v1.5.3
```
Comment (MicahParks) [inline go.mod:6]: A few bug updates I can discuss in further detail if you'd like. Primarily updates for RFC compliance.


* https://github.com/MicahParks/keyfunc/releases/tag/v1.5.3
* https://github.com/MicahParks/keyfunc/releases/tag/v1.5.2

**Comment by MicahParks** [comment]
@jensneuse I believe the PR now meets the requirements for a PR to this repository. The only intended changes are to update `keyfunc`.

**Comment by jensneuse** [comment]
> @jensneuse I believe the PR now meets the requirements for a PR to this repository. The only intended changes are to update `keyfunc`.

Ok, thank you.
We'll add tests for jwks through this branch and then merge it.
Putting it on our backlog.

**Comment by jensneuse** [comment]
Not stale. We'll merge this soon.

**Comment by jensneuse** [comment]
Thank you @MicahParks for the contribution!

**Pull request closed without merging**